inc/integrations/api/form-response-data.php

```diff
@@ -280,77 +289,118 @@ public function set_is_credential_error( $is_credential_error ) {
 	 * @since 2.1.7
 	 */
 	public function process_error_code() {
-		switch ( $this->response['code'] ) {
+		$this->add_reason( self::get_error_code_message( $this->response['code'] ) );
+	}
+
+	/**
+	 * Get the error message based on the error code.
+	 *
+	 * @param int $error_code The error code.
+	 * @return string
+	 * @since 2.2.3
+	 */
+	public static function get_error_code_message( $error_code ) {
+		$messages = array();
```

Comment:
Why is this an array? I only see $messages returning a single string. Is there a reason?

Also, you can use an array instead of a switch like this:

```
$error_messages = array(
	self::ERROR_MISSING_DATA => __( 'Essential data is missing: invalid Form id or protection.', 'otter-blocks' ),
	self::ERROR_MISSING_CAPTCHA => __( 'Captcha token is missing.', 'otter-blocks' ),
	// And so on
);

if ( ! isset( $error_messages[ $error_code ] ) ) {
	return "Expected error whatever message";
}

return $error_messages[ $error_code ];
```

Reply:
The main idea was to allow multiple error composition (not the case at the current complexity).

If we do more integration, a lot of errors will overlap. So instead of making a separate string for every error, we can compose like: `array( 'Provider Mailchimp', 'Invalid API Key')` or `array( 'Provider Sendinblue', 'Invalid API Key')`

Or course, we can concatenate the string and return a single one, but I liked the idea of an array, especially if we will create some more complex error handling.

Reply:
Until we do that we can just keep it simple. Not the most important thing.